common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java

```diff
@@ -317,11 +317,11 @@ public AcquireTokenResult acquireTokenSilent(
         // other tenants. Those tokens will be 'sparse', meaning that their AT/RT will not be loaded
         final ICacheRecord fullCacheRecord = cacheRecords.get(0);
 
-        if (accessTokenIsNull(fullCacheRecord)
+        if ((accessTokenIsNull(fullCacheRecord)
                 || refreshTokenIsNull(fullCacheRecord)
                 || parametersWithScopes.isForceRefresh()
                 || !isRequestAuthorityRealmSameAsATRealm(parametersWithScopes.getAuthority(), fullCacheRecord.getAccessToken())
-                || !strategy.validateCachedResult(authScheme, fullCacheRecord)) {
+                || !strategy.validateCachedResult(authScheme, fullCacheRecord)) && !accessTokenNeedsRefresh(fullCacheRecord)) {
```

Comment:
If refresh is needed because of refresh_in, but request fails, I think we should still return the underlying unexpired AT.

Reply:
Outdated